### PR TITLE
Deprecation warnings in GitHub workflows

### PR DESCRIPTION
**Pull request by cubeddu**
Issue [#156](https://github.com/aws/graph-explorer/issues/156)

Updated build_docker.yml and unit.yml to address annotations from deprecation warnings.

This PR addresses issue #156 by updating the build_docker.yml and unit.yml files to use the latest versions of the actions and address annotations from deprecation warnings.

In build_docker.yml, we updated the `actions/checkout` action to the latest version.

In unit.yml, we updated the `actions/setup-node` action to version 2 and removed the `node-version` option, which is now deprecated. We also updated the `codecov/codecov-action` action to version 3 and removed the `directory` and `files` options, which are not necessary for the default configuration.

Links to the updated files:
- [.github/workflows/build_docker.yml](https://github.com/aws/graph-explorer/compare/main...cubeddu:graph-explorer:bug/actions#diff-c7f993a522ae12469fcfb15fc43197a8be4d9ef9e1350d98e775b2603eec9bd0)
- [.github/workflows/unit.yml](https://github.com/aws/graph-explorer/compare/main...cubeddu:graph-explorer:bug/actions#diff-e17ac7ab478a18e9dac875e191cc3e0d754fab62eddddfb05d4619495be927d9)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build_docker.yml [hunk 1]
```diff
@@ -31,18 +31,19 @@ jobs:
...
         id: package-version
+        if: runner.os != 'Windows'
 
       - name: Get image tag
         id: get-image-tag
         run: |
           if ${{ github.event_name == 'workflow_dispatch' }} ; then
             if ${{ inputs.image_tag == 'release'}}; then
-              echo "image_tag=${{ steps.package-version.outputs.version }}" >> $GITHUB_OUTPUT
+              echo "image_tag=${{ steps.package-version.outputs.version }}" >> $GITHUB_ENV
```
Comment (michaelnchin) [inline .github/workflows/build_docker.yml:41]: The `GITHUB_OUTPUT` env variable should be left alone here and elsewhere in this workflow - this is the replacement for the old `"::set-output` command and should not trigger warnings.

The deprecation warnings are coming from the third-party actions `tyankatsu0105/read-package-version-actions@v1` ([on L30](https://github.com/aws/graph-explorer/blob/main/.github/workflows/build_docker.yml#L30)) and `aws-actions/configure-aws-credentials@v1` ([on L49](https://github.com/aws/graph-explorer/blob/main/.github/workflows/build_docker.yml#L49)). 

We should be able to upgrade the latter to  `aws-actions/configure-aws-credentials@v2` without issue (see [here](`aws-actions/configure-aws-credentials@v1)). The first action for reading the JS package version will need a replacement library, as the author has not updated it in several years.

**Review by michaelnchin — CHANGES_REQUESTED** [review]

File: .github/workflows/build_docker.yml (hunk 1 above)
Comment (cubeddu) [reply .github/workflows/build_docker.yml]: @michaelnchin 
I went ahead a reverted the changes made to the Get image tag step, updated those actions, and replaced the package version with what looks more promising action.

File: .github/workflows/build_docker.yml [hunk 2]
```diff
@@ -57,7 +55,7 @@ jobs:
 
       - name: Login to Amazon ECR
         id: login-ecr-public
-        uses: aws-actions/amazon-ecr-login@v1
+        uses: karlonovak/amazon-ecr-login@v2.0.0
```
Comment (michaelnchin) [inline .github/workflows/build_docker.yml:58]: I would prefer that we stay on the AWS-maintained version of this action - could you elaborate on the reason for changing this? 

I believe the latest versions of the original action shouldn't trigger any of the deprecation warnings, the requisite changes were made to remove `set-output` (see [here](https://github.com/aws-actions/amazon-ecr-login/issues/358#issuecomment-1295750476)) and upgrade to Node 16 (see [here](https://github.com/aws-actions/amazon-ecr-login/commit/29516697cce506511bbd2ab14eec27dcfb5626b9)).

**Review by michaelnchin — CHANGES_REQUESTED** [review]

File: .github/workflows/build_docker.yml (hunk 1 above)
Comment (michaelnchin) [reply .github/workflows/build_docker.yml]: Awesome, thank you for the updates @cubeddu 👍

File: .github/workflows/build_docker.yml (hunk 2 above)
Comment (cubeddu) [reply .github/workflows/build_docker.yml]: not a problem I reverted it back to stay within scope.

**Review by michaelnchin — DISMISSED** [review]
Looks good, thanks @cubeddu !

**Review by michaelnchin — APPROVED** [review]

**Pull request merged**